### PR TITLE
[ISSUE #1917] Specify executor to run CompletionStage

### PR DESCRIPTION
**Pull request by rushsky518**
## What is the purpose of the change

Specify executor to run CompletionStage

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

**Comment by rushsky518** [comment]
#1917

File: broker/src/main/java/org/apache/rocketmq/broker/processor/SendMessageProcessor.java [hunk 1]
```diff
@@ -311,9 +311,9 @@ public boolean rejectRequest() {
                                                                             SendMessageContext sendMessageContext,
                                                                             ChannelHandlerContext ctx,
                                                                             int queueIdInt) {
-        return putMessageResult.thenApply((r) ->
-            handlePutMessageResult(r, response, request, msgInner, responseHeader, sendMessageContext, ctx, queueIdInt)
-        );
+        return putMessageResult.thenApplyAsync((r) ->
```
Comment (duhenglucky) [inline broker/src/main/java/org/apache/rocketmq/broker/processor/SendMessageProcessor.java:314]: how about  change to: 

        putMessageResult.thenAcceptAsync(result -> {
            RemotingCommand remotingCommand = handlePutMessageResult(result, response, request, msgInner, responseHeader, sendMessageContext, ctx, queueIdInt);
            if (remotingCommand != null) {
                doResponse(ctx, request, remotingCommand);
            }

        }, this.brokerController.getSendMessageExecutor());

        return null;

Comment (rushsky518) [reply broker/src/main/java/org/apache/rocketmq/broker/processor/SendMessageProcessor.java]: i have changed

Comment (rushsky518) [reply broker/src/main/java/org/apache/rocketmq/broker/processor/SendMessageProcessor.java]: i tried to change it, but found that
`SendMessageProcessor#processRequest` 
is invoked in 
`SendMessageProcessorTest#assertPutResult`, 
that means a returned response is necessary

**Review by duhenglucky — DISMISSED** [review]

File: broker/src/main/java/org/apache/rocketmq/broker/processor/SendMessageProcessor.java [hunk 2]
```diff
@@ -79,7 +79,7 @@ public RemotingCommand processRequest(ChannelHandlerContext ctx,
 
     @Override
     public void asyncProcessRequest(ChannelHandlerContext ctx, RemotingCommand request, RemotingResponseCallback responseCallback) throws Exception {
-        asyncProcessRequest(ctx, request).thenAcceptAsync(responseCallback::callback, this.brokerController.getSendMessageExecutor());
+        asyncProcessRequest(ctx, request).thenAccept(responseCallback::callback);
```
[the remark this answers is not in the document]
Comment (rushsky518) [reply broker/src/main/java/org/apache/rocketmq/broker/processor/SendMessageProcessor.java:82]: `thenApplyAsync` is invoked in `asyncProcessRequest`, so `thenAcceptAsync` here is unnecessary

[the remark this answers is not in the document]
Comment (xiangwangcheng) [reply broker/src/main/java/org/apache/rocketmq/broker/processor/SendMessageProcessor.java:82]: OK, I think `thenAccept` and `thenAcceptAsync` are both OK, no big difference. The only difference between  `thenAccept` and `thenAcceptAsync` is the task in `thenAcceptAsync` is always executed in a different thread from a thread pool, while the task in `thenAccept` may be executed in the main thread, depending on some circumstances.

**Review by xiangwangcheng — DISMISSED** [review]

**Comment by xiangwangcheng** [comment]
LGTM. Can u resolve the conflicts above？

**Comment by rushsky518** [comment]
> resolve

resolved

**Comment by rushsky518** [comment]
> LGTM. Can u resolve the conflicts above？
resolved

**Pull request closed without merging**